showcase/src/elm/UrlGenerator.elm

```diff
@@ -0,0 +1,513 @@
+module UrlGenerator exposing (fromSourceCode)
+
+import Parser exposing ((|.), (|=), DeadEnd, Parser, Problem(..), Step(..))
+
+
+
+-- Can also be put into Utils
+-- but it might get pretty complicated so it's probably better to be put in its own file.
+
+
+fromSourceCode : String -> String -> String
```

Comment:
> Function Signature Is Error Prone

I like the idea of creating a custom `Version` type , and I think an opaque type would probably be the best way to go to enforce this. One issue that could come up would be the necessity to JSON decode the flags, since a custom type wouldn't be possible to send over from Javascript. Then we'd have to deal with the possibility of decoding failures! Though we could also convert it from the `Maybe String` during `Router.init`. 

One solution I've seen other people use (especially the elm-ui library) is unnamed type aliases (or whatever it's called). So the function could be 
```elm 
fromSourceCode : { version : String, code: String } -> String
```

This makes it much easier to read imo, though doesn't guarantee that the user would mix up version and code, which shouldn't happen too much, given how obvious it is. One downsize is that it makes currying a lot more difficult, but I don't think that would happen very much with this function. 

What do you think? I'm going to go with the "unnamed type aliases" style and see how it works out. If the custom `Version` type ends up being better, it won't take too much time to change. 

>  I would recommend that you add a comment to this function

That's a great idea! I'm thinking I could get to that after the PR gets merged.

Reply:
I'm also very new in the realm of tests, so I'll definitely appreciate your help!